src/sentry/api/serializers/models/releaseactivity.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+from typing import Any, Mapping, MutableMapping, Sequence, TypedDict
+
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.models import Group
+from sentry.models.releaseactivity import ReleaseActivity
+from sentry.utils.json import JSONData
+
+
+class ReleaseActivityResponse(TypedDict):
+    id: int
+    type: str
+    data: Mapping[str, JSONData]
+    date_added: datetime
+
+
+@register(ReleaseActivity)
+class ReleaseActivitySerializer(Serializer):
+    def get_attrs(
+        self, item_list: Sequence[ReleaseActivity], user: Any, **kwargs: Any
+    ) -> MutableMapping[Any, Any]:
+        def _update_data(d: MutableMapping[str, Any]):
+            if d.get("group_id"):
+                try:
+                    d["group"] = Group.objects.get(id=d.get("group_id"))
+                except Group.DoesNotExist:
+                    d["group"] = None
+                d.pop("group_id", None)
+                d["group"] = serialize(d["group"])
+            return d
+
+        return {activity: _update_data(activity.data) for activity in item_list}
```

Comment:
Mutating the data seems like a risky move, can we just include this in a `group` key or something? If someone happened to save a model after this it'd mess things up.

Alternatively you can just copy data here and use that

Reply:
Yeah, good call out. I'll make a copy of the data instead of mutating.